### ISSUE TITLE
[Communities] Enable Board MVP and enhance Create Community flow

### ISSUE THREAD

**Issue by ori-near** [near/neardevhub-widgets#156]
**Problem**
Some users run into issues with creating communities after spending significant time on an extensive form. Additionally, communities lack an effective method to organize posts according to their workflows, making it difficult to visualize progress. To address these issues, we need to enhance the community creation experience and introduce a basic customizable Kanban board feature to organize posts.

**User Story**
As a user, I want a smooth community creation process without error and the ability to organize posts for my community to enhance visibility.

**Acceptance Criteria**

**1. Creating Communities**
- [ ] Clicking the "Create Community" button displays a concise form above the community list
- [ ] The form section is called "Community Information" and it display critical fields: Name, Description, URL Handle (Changed from Handle), and Tag (allow up to 30 characters).
- [ ] The form displays "Launch" and "Cancel" buttons. A user must enter all fields in order to click "Launch" and proceed.
- [ ] While the form is active, the "Create Community" button is hidden. Canceling the form restores the button.
- [ ] Once a user enters the required fields, clicks the "Launch" button and proceeds with the transaction confirmation, the form disappears and the new community appears at the end or beginning of the communities page
- [ ] By default, the user is set as the default admin and the placeholder images are set as the logo and banner

**2. Viewing Community (After Creation)**
- [ ] Clicking the newly created community reveals:
  - [ ] Top header contains "Configure Community" button for settings
  - [ ] Navigation tabs: Activity, Teams, GitHub, and Board (need to remove old sponsorships page)
  - [ ] Sidebar description by default contains placeholder text: "This is a sample text about your community. Edit this in the configure community page." Remove the label or anything else. Only text description should show.
Note: Group moderatos section will be updated as part of https://github.com/near/neardevhub-widgets/issues/241

**3. Configuring Board Tab**
- [ ] On the Board tab, all user see a similar message to the existing GitHub tab: "This community does not have a board set up."
- [ ] Admins will see a "Create Board" button below the message. Clicking on it leads to a configuration form, similar to the GitHub tab.
- [ ] Admins can configure the page name and enable up to 10 columns in the Kanban board
- [ ] Adnubs can choose which fields to show/hide in the Kanban cards (e.g. post title, post author, post type, post labels, # replies)
- [ ] Admin can customize column name, description, and filtering criteria
- [ ] Filtering criteria should display only DevHub posts with specific labels in each column

**4. Configuring Community Page**
- [ ] Admin has the ability to edit all critical fields that were displayed in the create flow.
- [ ] Admin has the ability to edit all other sections that were not displayed in the create flow (e.g. About, Access Control, Wiki Page 1, Wiki Page2)

**Comment by ailisp** [comment]
@ori-near We need some clarification with Table View. Is it like github timeline? It would be nice to have the design for both table view and kanban view. Not urgent, we have enough tasks to work with before you're back next week.

**Comment by near-akaia-root** [comment]
@ori-near I propose to change the ticket title to "[Communities] Customizable project management"

**Comment by ori-near** [comment]
@ailisp Carina and I already discussed this. The table view is similar to this: https://github.com/orgs/near/projects/60/views/2

**Comment by carina-akaia** [comment]
The task takes longer than initially expected for two major reasons:
1. It's hard to give a precise estimate with near-social-vm due to its limited capabilities and available tooling. The solution is described in #118 
2. It seems that the time required to implement the contract side update wasn't properly assessed. I suggest to always create a separate ticket for contract updates and consider them separately.

I'd like to offload more logic to the contact side, taking sufficient part of the complexity away from frontend

**Comment by carina-akaia** [comment]
## Current progress
Contract side: the necessary minimum is done in https://github.com/near/neardevhub-contract/pull/45, but I'd like to implement project view configuration functionalities instead of dealing with raw JSON
Frontend side: everything except project view configuration is done in #218

**Comment by ailisp** [comment]
Let's keep contract side necessary minimum. Dealing with raw json in frontend will takes less time than refine it in contract side. Because raw json you have set of libraries that dealing with github schema/configuration, which can be reused for project view configuration. Also the format is subject to change, a structured project view configuration in contract side will require a change in contract every time

**Comment by ailisp** [comment]
Hi @carina-akaia How is the progress? Looks you have some updates in contract and widget, but both are still in draft stage. Let me know if there's any blockers!

**Comment by carina-akaia** [comment]
Hi @ailisp. At this moment I only need to deal with kanban view columns, for some reason they break when they're presented as an array rather than an object of entries indexed by ids ( as it's done for github views ). I don't see anything besides it left, but I'll conduct one final manual testing flow once I finish and fix any issues I may find in the process. I'll do my best to open the frontend PR for review tonight

**Comment by ori-near** [comment]
Hi @ailisp / all – Due to several weeks of technical challenges and delays, along with identified UX concerns during the community creation process, @carina-akaia and I refined the scope of this ticket. 

**Initial Scope** 
Originally, the plan was to introduce a board framework that allowed users to enable/disable multiple boards and configure advanced settings such as the Kanban board and tables views.

**Simplified Scope**
For the refined MVP, we will deliver just a single default Kanban board that users can customize, similar to the first GitHub tab that we launched. Additionally, we will address issues in the community creation process.

This shift will allow us to deliver the basic board functionality, address the user experience gaps, and make it easier for other developers to build on future enhancements.
